src/sdk/namespace_fs.js

```diff
@@ -1111,6 +1112,27 @@ class NamespaceFS {
         }
     }
 
+    // handle read of specific part number from a multipart upload
+    // see _handle_complete_multipart_size()
+    _handle_read_part_number(params, res, stat) {
+        if (params.part_number) {
+            const multipart_size = Number(stat.xattr?.[XATTR_MULTIPART_SIZE]);
+            if (multipart_size > 0) {
+                params.start = multipart_size * (params.part_number - 1);
+                // last part may be smaller
+                params.end = Math.min(multipart_size * params.part_number, stat.size);
+                // TODO s3_get_object has a similar handling which better be written once
+                const content_range = `bytes ${params.start}-${params.end - 1}/${stat.size}`;
+                res.statusCode = 206;
+                dbg.log0(params.start, params.end, stat.size, multipart_size, content_range);
+                res.setHeader?.('Content-Range', content_range);
```

Comment:
setting properties on res should not be on this file but on the s3 layer and probably should be written once (as written in the TODO :), check s3_get_object.js file)

Reply:
perhaps add this to a new function in `http_utils.set_response_range()` which can use the existing `http_utils.format_http_ranges()` function